### PR TITLE
Update dependencies to latest versions

### PR DESCRIPTION
**Pull request by matsev**
The `rimraf` dependency is [out of date](https://david-dm.org/raszi/node-tmp), and so are some of the `devDependencies`. This PR updates the dependencies to the latest version.

### CODE REVIEW

**Comment by matsev** [comment]
I get the same [failing tests](https://travis-ci.org/raszi/node-tmp/jobs/583686782) when running the tests on the master branch on my local computer (i.e. without my changes), so I am not convinced that they origin from this PR. My best guess is that they origin from (transitive) dependency updates since `node-tmp` does not have [package-lock.json](https://nodejs.dev/the-package-lock-json-file) that set them in stone.

**Comment by raszi** [comment]
Thank you @matsev for your PR. Unfortunately we cannot simply update the dependencies like that because we are still supporting Node v6.

**Comment by silkentrance** [comment]
@raszi since a few releases back we only support node >7.
This is due to the fact that SIGINT does not work with node 6 and the respective tests will just fail.

@matsev I have checked out your PR and ran the tests locally. It seems as if this is a race condition. After having run the tests locally against node v12.10.0, I encountered the same issue. However, checking the temporary path tells me that the temporary directory had been removed as expected. Yet, during test evaluation, the temporary directory still exists. I will look into this, maybe we need to introduce some, yikes, wait time before checking whether the directory does no longer exist.

**Comment by silkentrance** [comment]
And this happens when you swallow up exceptions, my bad:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type function
    at Object.rmdirSync (fs.js:754:13)
    at Array._cleanupCallback (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:18611)
    at _garbageCollector (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:19234)
    at process._tmp$safe_listener (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:23541)
    at process.emit (events.js:214:15)
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type function
    at Object.rmdirSync (fs.js:754:13)
    at Array._cleanupCallback (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:18611)
    at _garbageCollector (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:19234)
    at process._tmp$safe_listener (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:23541)
    at process.emit (events.js:214:15)
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type function
    at Object.rmdirSync (fs.js:754:13)
    at Array._cleanupCallback (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:18611)
    at _garbageCollector (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:19234)
    at process._tmp$safe_listener (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:23541)
    at process.emit (events.js:214:15)
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type function
    at Object.rmdirSync (fs.js:754:13)
    at Array._cleanupCallback (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:18611)
    at _garbageCollector (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:19234)
    at process._tmp$safe_listener (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:23541)
    at process.emit (events.js:214:15)
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type function
    at Object.rmdirSync (fs.js:754:13)
    at Array._cleanupCallback (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:18611)
    at _garbageCollector (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:19234)
    at process._tmp$safe_listener (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:23541)
    at process.emit (events.js:214:15)
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type function
    at Object.rmdirSync (fs.js:754:13)
    at Array._cleanupCallback (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:18611)
    at _garbageCollector (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:19234)
    at process._tmp$safe_listener (/var/workspaces/01-coldrye/01-collaboration/node-tmp/lib/tmp.js:9:23541)
    at process.emit (events.js:214:15)
```

**Comment by silkentrance** [comment]
So we have multiple errors here:

1. the remove callback will pass the next callback to also rmdirSync, which does not work this way
2. and there is definitely a race condition somewhere (i had introduced console.log statements in the source and the tests no longer failed, and after having removed these log statements, the tests will started failing again)

**Comment by silkentrance** [comment]
It seems that the behaviour of node fs changed in at least 12.0.x. With the introduction of the experimental option `recursive` it will no longer remove an empty directory, which seems to be a bug.
Still, we must rely on the platform to not delete any non empty directories.

I am still figuring out on how we could achieve this without reimplementation of recursive deletes.

**Comment by silkentrance** [comment]
The actual failure lies in node v12.10.x which introduced the, seemingly, non optional second argument to fs.rmdirSync, which expects it to be a configuration object.

And while we definitely should fix our software, it stands to reason on why the peops over at node would introduce an optional formal parameter... while they could always check for additional parameters via the arguments list.

**Comment by silkentrance** [comment]
In addition, it seems as if node v12.10.x is now preventing from removing empty directories if the experimental recursive option is missing.

**Comment by silkentrance** [comment]
@matsev I have some changes that I would like to include with your PR. Can you please check the appropriate option for the PR?

**Comment by matsev** [comment]
@silkentrance I guess that you are referring to the [Allow edits from maintainers](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) checkbox? As far as I can tell it is checked so you should be good to go.

**Comment by silkentrance** [comment]
@matsev I have rebased you PR to the latest master. Let's see whether this will build without errors.

**Comment by silkentrance** [comment]
@matsev build is still failing with latest version of node. need to investigate.

**Comment by silkentrance** [comment]
@matsev I somehow have misplaced the fix I made for this issue as mentioned in https://github.com/raszi/node-tmp/issues/199#issuecomment-538660745.

Now, which machine has it...

**Comment by silkentrance** [comment]
Finally I managed to get my head around this. Turned out that I had to actually provide my credentials when pushing to your branch. In the past I thought this to be a mistake of mine. Damn.

**Comment by silkentrance** [comment]
Working now. Merging now.

**Comment by silkentrance** [comment]
Fixed with #198.

File: lib/tmp.js [hunk 1]
```diff
@@ -610,43 +617,33 @@ function _is_legacy_listener(listener) {
...
     const lstnr = listeners[i];
     /* istanbul ignore else */
-    if (lstnr.name === SIGINT_LISTENER_NAME) {
+    if (lstnr.name === '_tmp$sigint_listener') {
       existingListeners.push(lstnr);
       process.removeListener(SIGINT, lstnr);
     }
   }
-  process.on(SIGINT, function _tmp$sigint_listener() {
+  process.on(SIGINT, function _tmp$sigint_listener(doExit) {
```
Comment (nomcopter) [inline lib/tmp.js:630]: This variables `doExit` will just be the string `SIGINT`. Meaning this will _always_ exit on line 644.

Comment (silkentrance) [reply lib/tmp.js:630]: This has been resolved as we do non longer add a SIGINT handler.

Regardless, the logic behind this was to postpone exiting to the very last sigint handler, supporting multiple legacy versions of tmp here.

**Pull request merged**